frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditImage/UploadImageInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react'
+import { ControllerRenderProps } from 'react-hook-form'
+import { forwardRef } from '@chakra-ui/react'
+import imageCompression from 'browser-image-compression'
+
+import {
+  MAX_UPLOAD_FILE_SIZE,
+  MB,
+  VALID_UPLOAD_FILE_TYPES,
+} from '~shared/constants/file'
+
+import Attachment from '~components/Field/Attachment'
+
+import { EditImageInputs } from './EditImage'
+
+type UploadImageInputProps = Pick<
+  ControllerRenderProps<EditImageInputs, 'attachment'>,
+  'name' | 'onChange' | 'value'
+> & {
+  onError?: (errMsg: string) => void
+}
+
+export const UploadImageInput = forwardRef<UploadImageInputProps, 'div'>(
+  ({ name, onChange, value, onError }, ref): JSX.Element => {
+    useEffect(() => {
+      return () => {
+        if (!value?.srcUrl) return
+        URL.revokeObjectURL(value.srcUrl)
+      }
+    }, [value?.srcUrl])
+
+    const handleChange = (file?: File) => {
+      if (!file) return onChange({ file, srcUrl: undefined })
+
+      if (file.name.split('.').pop() === 'gif') {
+        // No compression for gifs
```

Comment:
Why do we even care for this? Do gif file become unusable after compression? Or we found they don't compress well, so compression is not needed?

I'm thinking that unless gifs somehow become corrupted with compression, I'd remove this conditional code have a single code flow for all images.

Just for reference png files are internally compressed with zlib, so they likely don't compress very well with one extra round too.

Reply:
gifs can't be compressed, they turn into a single static image (first frame compressed and returned)

Reply:
Oh got it! Apologies, I assumed this was standard file compression, but this is looking at image data, not random bits.